universe/interface.go

```diff
@@ -262,11 +262,24 @@ func (m *Leaf) SmtLeafNode() *mssmt.LeafNode {
 	return mssmt.NewLeafNode(m.RawProof, amount)
 }
 
-// LeafKey is the top level leaf key for a universe. This will be used to key
-// into a universe's MS-SMT data structure. The final serialized key is:
-// sha256(mintingOutpoint || scriptKey). This ensures that all
-// leaves for a given asset will be uniquely keyed in the universe tree.
-type LeafKey struct {
+// LeafKey is an interface that allows us to obtain the universe key for a leaf
+// within a universe.
+type LeafKey interface {
```

Comment:
Very nice solution! Awesome to be able to leave behind the universe collision issue for new trees :100: